src/geanyobject.c

```diff
@@ -380,3 +389,13 @@ GObject *geany_object_new(void)
 {
 	return g_object_new(GEANY_OBJECT_TYPE, NULL);
 }
+
+/** Get the global GeanyObject instance
+ *
+ * @return @transfer{none} The instance
+ */
+GEANY_API_SYMBOL
+GObject *geany_object_get_instance(void)
```

Comment:
meh.  If *really* you can't use a custom `signal_connect()` (why, as anyway you need to fake the object?), I'd really rather add the object to `GeanyData` or something like that than a `get_instance()` method (like main widgets, etc.).

It's even nicer API-wise IMO to do `self.geany_data.object.connect(...)` than `Geany.Object.get_instance().connect(...)`